src/status_im/contexts/wallet/create_account/new_keypair/keypair_name/view.cljs

```diff
@@ -2,44 +2,64 @@
     [quo.core :as quo]
     [react-native.core :as rn]
-    [reagent.core :as reagent]
     [status-im.common.floating-button-page.view :as floating-button-page]
+    [status-im.common.validators :as validators]
     [status-im.contexts.wallet.create-account.new-keypair.keypair-name.style :as style]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
 (def keypair-name-max-length 15)
+(def keypair-name-min-length 5)
 
-(defn view
-  []
-  (let [keypair-name (reagent/atom "")]
-    (fn []
-      (let [customization-color (rf/sub [:profile/customization-color])]
-        [rn/view {:style {:flex 1}}
-         [floating-button-page/view
-          {:header [quo/page-nav
-                    {:icon-name           :i/arrow-left
-                     :on-press            #(rf/dispatch [:navigate-back])
-                     :accessibility-label :top-bar}]
-           :footer [quo/bottom-actions
-                    {:actions          :one-action
-                     :button-one-label (i18n/label :t/continue)
-                     :button-one-props {:disabled?           (or (zero? (count @keypair-name))
-                                                                 (> (count @keypair-name)
-                                                                    keypair-name-max-length))
-                                        :customization-color customization-color
-                                        :on-press            #(rf/dispatch [:wallet/new-keypair-continue
-                                                                            {:keypair-name
-                                                                             @keypair-name}])}
-                     :container-style  style/bottom-action}]}
-          [quo/text-combinations
-           {:container-style style/header-container
-            :title           (i18n/label :t/keypair-name)
-            :description     (i18n/label :t/keypair-name-description)}]
-          [quo/input
-           {:container-style {:margin-horizontal 20}
-            :placeholder     (i18n/label :t/keypair-name-input-placeholder)
-            :label           (i18n/label :t/keypair-name)
-            :char-limit      keypair-name-max-length
-            :auto-focus      true
-            :on-change-text  #(reset! keypair-name %)}]]]))))
+(def error-messages
+  {:length       (i18n/label :t/key-name-error-length)
+   :emoji        (i18n/label :t/key-name-error-emoji)
+   :special-char (i18n/label :t/key-name-error-special-char)})
+
+(defn view-internal
+  [{:keys [theme]}]
+  (let [[keypair-name set-keypair-name] (rn/use-state "")
+        customization-color             (rf/sub [:profile/customization-color])
+        [error set-error]               (rn/use-state false)]
+    [rn/view {:style {:flex 1}}
+     [floating-button-page/view
+      {:header [quo/page-nav
+                {:icon-name           :i/arrow-left
+                 :on-press            #(rf/dispatch [:navigate-back])
```

Comment:
@mohsen-ghafouri we can, but is there a meaningful difference?

Reply:
as It doesn't depend on any parameter, it should be outside so that even if it's compared by reference, it will hold equal and not cause unnecessary re-rendering

refer to https://github.com/status-im/status-mobile/pull/19139#discussion_r1517637765